miden-crypto/src/dsa/ecdsa_secp256k1/mod.rs

```diff
@@ -0,0 +1,229 @@
+    utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable},
+};
+
+// CONSTANTS
+// ================================================================================================
+
+const SECRET_KEY_BYTES: usize = 32;
+const PUBLIC_KEY_BYTES: usize = 33; // we use the compressed format
+const SIGNATURE_BYTES: usize = 64;
+
+// SECRET KEY
+// ================================================================================================
+
+/// Secret key for ECDSA signature verification over secp256k1 curve.
+#[derive(Clone)]
+pub struct SecretKey {
+    inner: SigningKey,
+}
+
+impl SecretKey {
+    /// Generate a new random secret key using the OS random number generator.
+    #[cfg(feature = "std")]
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        let mut rng = k256::elliptic_curve::rand_core::OsRng;
+
+        Self::with_rng(&mut rng)
+    }
+
+    /// Generate a new secret key using the provided random number generator.
+    pub fn with_rng<R: CryptoRng + RngCore>(rng: &mut R) -> Self {
+        let signing_key = SigningKey::random(rng);
+        Self { inner: signing_key }
+    }
+
+    /// Get the corresponding public key for this secret key.
+    pub fn public_key(&self) -> PublicKey {
+        let verifying_key = self.inner.verifying_key();
+        PublicKey { inner: *verifying_key }
+    }
+
+    /// Sign a message (represented as a Word) with this secret key.
+    pub fn sign(&mut self, message: Word) -> Signature {
+        // Convert Word to bytes for signing
+        let message_bytes: [u8; 32] = message.into();
+
+        // Sign the message
+        let signature: k256::ecdsa::Signature = self.inner.sign(&message_bytes);
+
+        Signature { inner: signature }
+    }
+}
+
+// PUBLIC KEY
+// ================================================================================================
+
+/// Public key for ECDSA signature verification over secp256k1 curve.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PublicKey {
+    inner: VerifyingKey,
+}
+
+impl PublicKey {
+    /// Convert public key to a Word representation for use in Miden.
+    pub fn to_word(&self) -> Word {
+        self.to_commitment()
+    }
+
+    /// Verify a signature against this public key and message.
+    pub fn verify(&self, message: Word, signature: &Signature) -> bool {
+        let message_bytes: [u8; 32] = message.into();
+        self.inner.verify(&message_bytes, &signature.inner).is_ok()
+    }
+}
+
+impl SequentialCommit for PublicKey {
+    type Commitment = Word;
+
+    fn to_elements(&self) -> Vec<Felt> {
+        self.to_bytes().chunks(5).map(Felt::from_bytes_with_padding).collect()
+    }
+}
+
+// SIGNATURE
+// ================================================================================================
+
+/// ECDSA signature over secp256k1 curve
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Signature {
+    inner: k256::ecdsa::Signature,
+}
```

Comment:
Depending on what we decide with the Falcon signature, we may need to include the public key here as well. Or does EC recover help us here somehow?

Reply:
Yes, I think we can use EC recover, we only need to add 1 byte and then we are able to recover the public key from the signature.

Also one more thing, we probably want to use Keccak for hashing messages, right? As things currently stand, this PR uses the default hasher which is SHA256